project.go

```diff
@@ -181,18 +189,22 @@ func (rs project) Run(ctx context.Context, inp, out chan Dataset) (err error) {
 			}
 
 			if open {
-				result, err = result.Expand(curr)
-				if err != nil {
-					return err
+				if len(result) > 0 {
+					err := rs.hasDifferentLength(curr.At(0).Len(), result[0].Len())
```

Comment:
we can maintain resultLen (not current var named resultLen, but real length. i.e. number of lines) instead of calling it each time.

also, you might miss an error when first is dummy (len is -1), second is length 10, and curr is 20
```suggestion
					err := rs.hasDifferentLength(curr.Len(), resultLen)
```

Reply:
This isn't correct, because after the command `err := rs.hasDifferentLength(curr.Len(), resultWidth)`, there's the command `result = append(result, curr.(dataset)...)`.
i.e. the result's value is changing dynamicaly through the _for_ loop.

Reply:
result _width_ is changing. not result _length_